backend/service/service.go

```diff
@@ -612,3 +624,154 @@ func (s *APIService) UserMeTicketCredentialsGet(ctx context.Context) (openapi.Im
+    contextID, err := contextRegistry.CalculateContextID(&bind.CallOpts{}, contextString)
+    if err != nil {
+        logger.Err(err).Msg("Failed to calculate context ID")
+        return openapi.Response(http.StatusInternalServerError, nil), err
+    }
+
+	// hard code chain id as 1 (eth mainnet)
+	// chainID := fmt.Sprintf("%d", s.issuerChainID)
+	chainID := "1"
+
+	// hard code issuer key id for now
+	// issuerKeyID := fmt.Sprintf("%d", s.issuerKeyID)
+	// TODO: not sure if we should store this in hex or decimal
+	issuerKeyID := "0xc4525dA874A6A3877db65e37f21eEc0b41ef9877"
+
+	// create event
+	event, err := s.dbClient.Events.CreateEvent(ctx, events.CreateEventParams{
+		ID:          uuid.NewString(),
+		Name:        createEventRequest.Name,
+		Description: createEventRequest.Description,
+		Url:         createEventRequest.Url,
+		AdminCode:   createEventRequest.AdminCode,
+		ChainID:     chainID,
+		ContextID:   contextID.String(),
+		ContextString: contextString,
+		IssuerKeyID: issuerKeyID,
+		StartDate:   pgtype.Timestamptz{Time: createEventRequest.StartDate, Valid: true},
+		EndDate:     pgtype.Timestamptz{Time: createEventRequest.EndDate, Valid: true},
+	})
+	if err != nil {
+		logger.Err(err).Msg("Failed to create event")
+		return openapi.Response(http.StatusInternalServerError, nil), err
+	}
+
+	// add the user as an admin to the event_admins table
+	_, err = s.dbClient.EventAdmins.CreateEventAdmin(ctx, event_admins.CreateEventAdminParams{
+		EventID: event.ID,
+		UserID:  userID,
+	})
+	if err != nil {
+		logger.Err(err).Msg("Failed to create event admin")
+		return openapi.Response(http.StatusInternalServerError, nil), err
+	}
+
+	logger.Info().Str("eventID", event.ID).Msg("Created event")
+
+	return openapi.Response(http.StatusCreated, MarshalEvent(event)), nil
+}
+
+// EventsEventIdPut - Update an event
+func (s *APIService) EventsEventIdPut(ctx context.Context, eventId string, updateEventRequest openapi.UpdateEventRequest) (openapi.ImplResponse, error) {
+	logger := log.Ctx(ctx).With().Str("op", "EventsEventIdPut").Logger()
+	userID := util.GetUserIDFromContext(ctx)
+	userEmail := util.GetUserEmailFromContext(ctx)
+	if userID == "" || userEmail == "" {
+		return openapi.Response(http.StatusUnauthorized, nil), nil
+	}
+
+	_, err := s.dbClient.Events.GetEventByID(ctx, eventId)
+	if err != nil {
+		if err == pgx.ErrNoRows {
+			logger.Err(err).Msg("Event not found")
+			return openapi.Response(http.StatusNotFound, nil), nil
+		}
+		return openapi.Response(http.StatusInternalServerError, nil), err
+	}
+
+	// check if user is an admin of the event
+	_, err = s.dbClient.EventAdmins.GetEventAdminsByEventId(ctx, eventId)
+	if err != nil {
+		if err == pgx.ErrNoRows {
+			logger.Err(err).Msg("User is not an admin of the event")
+			return openapi.Response(http.StatusUnauthorized, nil), nil
+		}
+		return openapi.Response(http.StatusInternalServerError, nil), err
+	}
+
+	// TODO: not sure if we should update the context string
```

Comment:
let's disallow name change - or this would bring a lot more complexity to the protocol logic. e.g. user would received credential already would need to discard and get new ones